csrc/scheduler/greedy.cpp

```diff
@@ -307,32 +341,42 @@ class CompileTimeChecker : private IterVisitor {
     // For the scatter in and out tensors, the scatter dimension must
     // not be parallelized with TID.
     auto out_tv = ir_utils::getTvOutput(scatter);
-    checkConstrainedTv(out_tv, {constrained_out_logical_dim});
+    checkDomainConstraints(
+        out_tv->domain()->initialLoop(), {constrained_out_logical_dim});
 
     // In addition, the index and src tensors are not allowed to use
     // TID with the scatter dim. Their logical domains are not mapped
     // with the logical domains of the input and output tensors, so
     // they need to be checked separately.
-    checkConstrainedTv(
-        scatter->index()->as<TensorView>(), {constrained_out_logical_dim});
+    checkDomainConstraints(
+        TensorDomain::noReduction(
```

Comment:
```suggestion
        TensorDomain::noReductions(
```
**[Suggested by AI]** **Summary of the change**

The build was failing because the code was calling a method that does not exist:

• Old call: `TensorDomain::noReduction(`  
• Correct call: `TensorDomain::noReductions(`  (note the extra “s”)

By replacing `noReduction` with the correct method name `noReductions`, the compiler can now find the function, resolving the build error. No other logic or behavior was modified; this is purely a typo/rename fix.